Allow Komodo's password storage to use an empty "" password (bug 80724).
Index: toolkit/components/passwordmgr/src/storage-Legacy.js
===================================================================
RCS file: /cvsroot/mozilla/toolkit/components/passwordmgr/src/storage-Legacy.js,v
retrieving revision 1.27
diff -d -u -8 -p -r1.27 storage-Legacy.js
--- toolkit/components/passwordmgr/src/storage-Legacy.js	6 May 2008 07:45:35 -0000	1.27
+++ toolkit/components/passwordmgr/src/storage-Legacy.js	1 Nov 2008 17:50:22 -0000
@@ -1166,18 +1166,18 @@ LoginManagerStorage_legacy.prototype = {
             [password, userCanceled] =
                 this._decrypt(login.wrappedJSObject.encryptedPassword);
 
             // Probably can't hit this case, but for completeness...
             if (userCanceled)
                 break;
 
             // If decryption failed (corrupt entry?) skip it.
-            // Note that we allow password-only logins, so username con be "".
-            if (username == null || !password)
+            // Note that we allow both the username and password to be "".
+            if (username == null || password == null)
                 continue;
 
             // We could set the decrypted values on a copy of the object, to
             // try to prevent the decrypted values from sitting around in
             // memory if they're not needed. But thanks to GC that's happening
             // anyway, so meh.
             login.username = username;
             login.password = password;
